pandas/core/reshape/reshape.py

```diff
@@ -600,6 +600,27 @@ def stack_multiple(frame, level, dropna=True):
     return result
 
 
+def _stack_multi_column_index(columns):
+    if len(columns.levels) <= 2:
+        return columns.levels[0]._rename(name=columns.names[0])
+    levs = []
+    for lev, level_codes in zip(columns.levels[:-1], columns.codes[:-1]):
+        if -1 in level_codes:
+            lev = np.append(lev, None)
+        levs.append(np.take(lev, level_codes))
+    # Remove duplicate tuples in the MultiIndex.
+    tuples = zip(*levs)
+    unique_tuples = (key for key, _ in itertools.groupby(tuples))
+    new_levs = zip(*unique_tuples)
+    return MultiIndex.from_arrays(
+        [
```

Comment:
maybe add a comment pointing back to this PR

Reply:
Done.